front-end/src/pages/Login.js

```diff
@@ -0,0 +1,137 @@
+	const [usernameValidation, setUsernameValidation] = useState(false);
+	const [passwordValidation, setPasswordValidation] = useState(false);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		if (username == "") {
+			setUsernameValidation(true);
+		} else if (password == "") {
+			setPasswordValidation(true);
+		}
+	}
+
+	const handleChangePassword = (event) => {
+		setPassword(event.target.value);
+		setPasswordValidation(false);
+	}
+
+	const handleChangeUsername = (e) => {
+		setUsername(e.target.value);
+		setUsernameValidation(false);
+
+	}
+
+	return (
+		<Grid container spacing={0} direction="row">
+			<Grid container item xs={6} spacing={0} className="col-item col-left">
+				<Grid container direction="column" style={backgroundStyle.style}>
+					<Grid container direction="column" justify="center" alignContent="center" className="subject-title">
+						<GreenTextTypography variant="h3">
+							Concious Cities
+            			</GreenTextTypography>
+					</Grid>
+					<Grid container direction="column" justify="center" alignContent="center" className="description-title">
+						<Grid item>
+							<TitleTextTypography variant="h4">shape</TitleTextTypography>
+						</Grid>
+						<Grid item>
+							<TitleText1Typography>YOUR CITY</TitleText1Typography>
+						</Grid>
+						<Grid item>
+							<TitleTextTypography variant="h5">in new ways</TitleTextTypography>
+						</Grid>
+					</Grid>
+				</Grid>
+			</Grid>
+
+			<Grid container item xs={6} spacing={0} className="col-item col-right">
+				<Grid container direction="column" justify="center" alignContent="center" className="login-inner">
+					<Grid container direction="row" justify="flex-start" alignContent="center" alignItems="center">
+						<Grid item className="logo-wrap">
+							<LogoIcon />
+						</Grid>
+						<Grid item className="login-margin">
+							<TitleLoginTypography>Login</TitleLoginTypography>
+						</Grid>
+					</Grid>
+					<form onSubmit={handleSubmit}>
+						<Grid container direction="column" justify="center" alignContent="center">
+							<Grid item className="input-login">
+								<TextInput
+									label="Username"
+									name="username"
+									onChange={handleChangeUsername}
+									color="green"
+								/>
+							</Grid>
+							{usernameValidation && (
+								<ValidatiionLoginString>
+									Please enter a username.
+								</ValidatiionLoginString>
+							)}
+							<Grid className="input-login">
+								<PasswordInput
+									label="Enter your password"
+									name="password"
+									handleChangePassword={handleChangePassword}
+									color="green"
+								/>
+							</Grid>
+							{passwordValidation && (
+								<ValidatiionLoginString>
+									Please enter a password.
+								</ValidatiionLoginString>
+							)}
+						</Grid>
+						<Grid container item justify="center" alignContent="center">
+							<Link className="create_account_link" href="/">
```

Comment:
do not use href. use single page application routing so that refresh of page is not triggered, with react router
https://reactgo.com/react-router-usehistory-hook/